cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringServerHandler.java

```diff
@@ -141,6 +143,52 @@ public void connectControlChannel(FullHttpRequest request, HttpResponder respond
                                    TetheringControlResponse[].class));
   }
 
+  /**
+   * Sends control commands to the client and receives program status updates from the client.
+   */
+  @POST
+  @Path("/tethering/channels/{peer}")
+  public void pollControlChannel(FullHttpRequest request, HttpResponder responder, @PathParam("peer") String peer)
+    throws IOException, NotImplementedException, BadRequestException, TopicNotFoundException {
+    checkTetheringServerEnabled();
+    PeerInfo peerInfo;
+    try {
+      peerInfo = store.getPeer(peer);
+    } catch (PeerNotFoundException e) {
+      responder.sendString(HttpResponseStatus.OK,
```

Comment:
it's still more correct to return a 404 like before, though it complicates the client logic a little bit, as the client needs to try to deserialize the response body to differentiate from some other type of not found problem.

Reply:
fixed